shared/mempool/generic_fifo_set.go

```diff
@@ -68,16 +60,17 @@ func (g *GenericFIFOSet[TIdx, TData]) Push(item TData) error {
 		front := g.queue.Front()
 		delete(g.set, g.indexerFn(front.Value.(TData)))
 		g.queue.Remove(front)
+		g.onRemove(item, g)
```

Comment:
I did not catch this...

Reply:
whenever there's an overflow, we need to compensate for the onAdd action we did earlier because effectively we are deleting an element from the data structures